src/main/java/com/thoughtworks/gauge/datastore/DataStoreFactory.java

```diff
@@ -5,6 +5,14 @@
  *----------------------------------------------------------------*/
 package com.thoughtworks.gauge.datastore;
 
+import static com.thoughtworks.gauge.GaugeConstant.ENABLE_MULTITHREADING_ENV;
+
+
+/**
+ * @deprecated DataStoreFactory is no longer valid. This API will throw an Exception in multithreaded execution.
+ * <p>Use specific data stores {@link SuiteDataStore, SpecDataStore or ScenarioDataStore } instead </p>
```

Comment:
[WARNING] /Users/has23/workspace/etc/gauge-java/src/main/java/com/thoughtworks/gauge/datastore/DataStoreFactory.java:13: warning - invalid usage of tag {@link SuiteDataStore,